andela/ah-lobos-backend
# #166789866 Setup API Documentation using swagger

#### What does this PR do?

- Setup API Documentation using swagger

#### Description of Task to be completed?

- add swagger-jsdoc and swagger-ui-express packages
- add swaggerDefinition for app title and host
- add options for swagger definition and locate the folder
- add swagger specification for it options
- add comments above the route being documented

#### How should this be manually tested?
- Clone the project
```
> git clone https://github.com/andela/ah-lobos-backend.git
```
- Install all packages
```
> npm install
```
- Run the application
```
> npm run dev
```
- Navigate to `/api-docs`

or check by clicking the following: [link](https://ah-lobos-backend-swagger.herokuapp.com/api-docs/#/)

#### Any background context you want to provide?
- Create `.env` file with the help of `example.env`
- The project structure has been changed

#### What are the relevant pivotal tracker stories?

[#166789866](https://www.pivotaltracker.com/story/show/166789866)

#### Screenshots (if appropriate)

![image](https://user-images.githubusercontent.com/31614647/60399210-69c28700-9b61-11e9-9fd5-1028fe61ab3c.png)

<img width="1658" alt="Screen Shot 2019-06-30 at 22 05 21" src="https://user-images.githubusercontent.com/31614647/60401734-e4050280-9b85-11e9-8a42-b37486a0f54e.png">

<img width="1658" alt="Screen Shot 2019-06-30 at 22 05 36" src="https://user-images.githubusercontent.com/31614647/60401736-ecf5d400-9b85-11e9-8a89-5683d8655d5d.png">



#### Questions:

- N/A
